src/modules/axis.js

```diff
@@ -1,26 +1,41 @@
 import i18n from '@dhis2/d2-i18n'
+import { AXIS_ID_COLUMNS, AXIS_ID_ROWS, AXIS_ID_FILTERS } from './layout/axis'
 import {
-    AXIS_ID_COLUMNS,
-    AXIS_ID_ROWS,
-    AXIS_ID_FILTERS,
-    AXIS_ID_YEAR_OVER_YEAR_SERIES,
-    AXIS_ID_YEAR_OVER_YEAR_CATEGORY,
-} from './layout/axis'
+    LAYOUT_TYPE_DEFAULT,
+    LAYOUT_TYPE_PIE,
+    LAYOUT_TYPE_YEAR_OVER_YEAR,
+    LAYOUT_TYPE_PIVOT_TABLE,
+} from './layoutTypes'
+import { getLayoutTypeByVisType } from './visTypeToLayoutType'
 
-const axisNames = {
-    [AXIS_ID_COLUMNS]: i18n.t('Series'),
-    [AXIS_ID_ROWS]: i18n.t('Category'),
-    [AXIS_ID_FILTERS]: i18n.t('Filter'),
-    [AXIS_ID_YEAR_OVER_YEAR_SERIES]: i18n.t('Series'),
-    [AXIS_ID_YEAR_OVER_YEAR_CATEGORY]: i18n.t('Category'),
+const getAxisNamesByLayoutType = layoutType => {
+    switch (layoutType) {
+        case LAYOUT_TYPE_DEFAULT:
+        case LAYOUT_TYPE_PIE:
+        case LAYOUT_TYPE_YEAR_OVER_YEAR:
+        default:
+            return {
+                [AXIS_ID_COLUMNS]: i18n.t('Series'),
+                [AXIS_ID_ROWS]: i18n.t('Category'),
+                [AXIS_ID_FILTERS]: i18n.t('Filter'),
+            }
+        case LAYOUT_TYPE_PIVOT_TABLE:
+            return {
+                [AXIS_ID_COLUMNS]: i18n.t('Columns'),
+                [AXIS_ID_ROWS]: i18n.t('Rows'),
+                [AXIS_ID_FILTERS]: i18n.t('Filter'),
+            }
+    }
 }
 
-export const getAxisName = axisId => {
-    const name = axisNames[axisId]
-
+export const getAxisNameByLayoutType = (axisId, layoutType) => {
+    const name = getAxisNamesByLayoutType(layoutType)[axisId]
     if (!name) {
         throw new Error(`${axisId} is not a valid axis id`)
     }
 
     return name
 }
+
+export const getAxisNameByVisType = (axisId, visType) =>
```

Comment:
Do we really need three new exported functions or can we get away with limiting the API surface area and exposing just one (the existing one)?

Reply:
So the idea is that since `layoutType` is whats actually used for the lookup but in most places we only have access to `visType`, the `getAxisNameByVisType` removes one step of the process (getting `layoutType` from `visType`). But in some cases (mostly regarding `YearOverYear`), we need to explicitly get the name by `layoutType`, hence the two exported functions.

Reply:
`getAxisNameByVisType` is now removed